src/core/server/plugins/plugin.test.ts

```diff
@@ -29,15 +29,15 @@ import { createPluginInitializerContext, createPluginStartContext } from './plug
 
 const mockPluginInitializer = jest.fn();
 const logger = loggingServiceMock.create();
-jest.mock(
+jest.doMock(
   join('plugin-with-initializer-path', 'server'),
   () => ({ plugin: mockPluginInitializer }),
   { virtual: true }
 );
-jest.mock(join('plugin-without-initializer-path', 'server'), () => ({}), {
+jest.doMock(join('plugin-without-initializer-path', 'server'), () => ({}), {
   virtual: true,
 });
-jest.mock(join('plugin-with-wrong-initializer-path', 'server'), () => ({ plugin: {} }), {
+jest.doMock(join('plugin-with-wrong-initializer-path', 'server'), () => ({ plugin: {} }), {
```

Comment:
Could we use doMock everywhere instead of the new `*.test.mocks.ts` files?

Reply:
We really need that pattern. Typescript has an open bug which is keep allowing us to build the tests as we are doing with the ts-jest https://github.com/Microsoft/TypeScript/issues/16166#issue-232515907

As the imports are not being hoisted, things like the following would work on `ts-jest` but not with `babel-jest`

```
const mockJobQueueClient = { getInfo: jest.fn() };
jest.doMock('../lib/job_queue_client', () => ({ jobQueueClient: mockJobQueueClient }));

import React from 'react';
import { mountWithIntl } from 'test_utils/enzyme_helpers';
import { ReportInfoButton } from './report_info_button';
```

So we need that pattern to enclose the jest mocks in an import that would also be hoisted on the top with `babel-jest` before the other imports